python/cugraph-pyg/cugraph_pyg/tests/int/test_int_cugraph.py

```diff
@@ -0,0 +1,221 @@
+
+    for node_type, num_nodes in data[0]["num_nodes_dict"].items():
+        vertex_offsets[node_type] = last_offset
+        last_offset += num_nodes
+
+        blank_df = cudf.DataFrame(
+            {
+                "id": range(
+                    vertex_offsets[node_type], vertex_offsets[node_type] + num_nodes
+                )
+            }
+        )
+        blank_df.id = blank_df.id.astype("int64")
+
+        pG.add_vertex_data(blank_df, vertex_col_name="id", type_name=node_type)
+
+    # Add the remaining vertex features
+    for i, (node_type, node_features) in enumerate(data[0]["node_feat_dict"].items()):
+        vertex_offset = vertex_offsets[node_type]
+
+        feature_df = cudf.DataFrame(node_features)
+        feature_df.columns = [str(c) for c in range(feature_df.shape[1])]
+        feature_df["id"] = range(vertex_offset, vertex_offset + node_features.shape[0])
+        feature_df.id = feature_df.id.astype("int64")
+
+        pG.add_vertex_data(feature_df, vertex_col_name="id", type_name=node_type)
+
+    # Fill in an empty value for vertices without properties.
+    pG.fillna(0.0)
+
+    # Add the edges
+    for i, (edge_key, eidx) in enumerate(data[0]["edge_index_dict"].items()):
+        node_type_src, edge_type, node_type_dst = edge_key
+        print(node_type_src, edge_type, node_type_dst)
+        vertex_offset_src = vertex_offsets[node_type_src]
+        vertex_offset_dst = vertex_offsets[node_type_dst]
+        eidx = [n + vertex_offset_src for n in eidx[0]], [
+            n + vertex_offset_dst for n in eidx[1]
+        ]
+
+        edge_df = cudf.DataFrame({"src": eidx[0], "dst": eidx[1]})
+        edge_df.src = edge_df.src.astype("int64")
+        edge_df.dst = edge_df.dst.astype("int64")
+        edge_df["type"] = edge_type
+
+        # Adding backwards edges is currently required in both
+        # the cuGraph PG and PyG APIs.
+        pG.add_edge_data(edge_df, vertex_col_names=["src", "dst"], type_name=edge_type)
+        pG.add_edge_data(
+            edge_df, vertex_col_names=["dst", "src"], type_name=f"{edge_type}_bw"
+        )
+
+    # Add the target variable
+    y_df = cudf.DataFrame(data[1]["paper"], columns=["y"])
+    y_df["id"] = range(vertex_offsets["paper"], vertex_offsets["paper"] + len(y_df))
+    y_df.id = y_df.id.astype("int64")
+
+    pG.add_vertex_data(y_df, vertex_col_name="id", type_name="paper")
+
+    # Construct a graph/feature store and loaders
+    feature_store, graph_store = to_pyg(pG)
+    sampler = CuGraphSampler(
+        data=(feature_store, graph_store),
+        shuffle=True,
+        num_neighbors=[10, 25],
+        batch_size=50,
+    )
+    loader = NodeLoader(
+        data=(feature_store, graph_store),
+        shuffle=True,
+        batch_size=50,
+        node_sampler=sampler,
+        input_nodes=("author", graph_store.get_vertex_index("author")),
+    )
+
+    return loader
+
+
+@pytest.fixture(scope="module")
+def loader_hetero_mag_multi_gpu(rmmc):
```

Comment:
For setting RMM pool on multiple gpus you have to do this when we create the dask cluster. 

See API here: https://docs.rapids.ai/api/dask-cuda/nightly/api.html#dask_cuda.LocalCUDACluster

This probably only creates stuff at the client layer.  I suggest possibly removing the RMM pool here

Reply:
I removed the rmm pool creation